packages/react/src/components/RadioCardGroup/index.tsx

```diff
@@ -0,0 +1,183 @@
+  const onRadioBlur = () => setFocusIndex(null);
+  const onRadioClick = (index: number) => {
+    const radio = inputs.current?.[index];
+
+    if (!radio) {
+      return;
+    }
+
+    radio.click();
+    radio.focus();
+  };
+
+  useEffect(() => {
+    if (typeof value === 'undefined') {
+      return;
+    }
+
+    setCurrentValue(value);
+  }, [value]);
+
+  const radioButtons = radios.map((radio, index) => {
+    const {
+      label,
+      disabled,
+      value: radioValue,
+      cardImgSrc,
+      cardIcon,
+      id,
+      className,
+      ...other
+    } = radio;
+    const isChecked = currentValue === radioValue;
+    const isFocused = focusIndex === index;
+
+    return (
+      <div className={classNames('Radio')} key={id}>
+        <Card
+          variant="simple"
+          className={classNames('Radio_card Radio__overlay', {
+            'Radio__overlay--focused': isFocused,
+            'Radio__overlay--checked': isChecked,
+            'Radio__overlay--disabled': disabled
+          })}
+          type={isChecked ? 'radio-checked' : 'radio-unchecked'}
+          onClick={() => onRadioClick(index)}
+        >
+          <input
+            type="radio"
+            name={name}
+            value={radioValue}
+            id={id}
+            ref={input => {
+              if (!input) {
+                return;
+              }
+
+              inputs.current.push(input);
+            }}
+            onFocus={() => onRadioFocus(index)}
+            onBlur={() => onRadioBlur()}
+            onChange={() => {
+              handleChange(radioValue);
+              onChange(radio, inputs.current?.[index]);
+            }}
+            disabled={disabled}
+            checked={isChecked}
+            {...other}
+          />
+          <CardContent>
+            <div className={classNames('Radio_card_icon_div')}>
+              {isChecked && (
+                <Icon
+                  className={classNames('Radio_card--icon')}
+                  type="check-circle"
+                />
+              )}
+            </div>
+
+            <div className={classNames('Radio_card_base')}>
+              <img
+                className={classNames('Radio_card_img')}
+                src={cardImgSrc}
+                alt=""
+              />
+              <label htmlFor={id} className={classNames('Radio_card_label')}>
+                {label}
+              </label>
+            </div>
+          </CardContent>
+        </Card>
+      </div>
+    );
+  });
+
+  // Hack to prevent ESLint from erroring about this variable not
+  // being used. We want to pull it from `props` to ensure it's
+  // not passed through to the radiogroup element.
+  void defaultValue;
+
+  // reset the input refs array
+  // refs get clobbered every re-render anyway and this supports "dynamic" radios
+  // (changing the number of radio buttons for example)
+  inputs.current = [];
+
+  return (
+    <div
+      className={classNames('Radio_card_group')}
```

Comment:
```suggestion
      className="RadioCardGroup"
```

Following our BEM guidelines, this should match the name of the component.